tools/scripts/ManifestBuilder.rb

```diff
@@ -0,0 +1,436 @@
+      `mkdir -p #{File.dirname json_destination}`
+
+      # Download the JSON file and PNG file
+      begin
+        verbose "Writing #{json_destination}"
+        objects['json'].get(response_target: json_destination)
+        verbose "Writing #{png_destination}"
+        objects['png'].get(response_target: png_destination)
+      rescue Aws::Errors::ServiceError => service_error
+        next <<-WARN
+There was an error retrieving #{name}.json and #{name}.png from S3:
+#{service_error}
+The animation has been skipped.
+        WARN
+      end
+
+      true
+    end
+    download_progress_bar.finish unless download_progress_bar.nil?
+
+    @warnings.each {|warning| warn "#{bold 'Warning:'} #{warning}"}
+
+    info <<-EOS.unindent
+      Library downloaded to #{DOWNLOAD_DESTINATION}
+
+        #{dim 'd[ o_0 ]b'}
+    EOS
+
+    if @options[:spritelab] && @options[:upload_to_s3]
+      info "Uploading file to S3"
+      AWS::S3.upload_to_bucket(
+        DEFAULT_S3_BUCKET,
+        "manifests/spritelabCostumeLibrary.json",
+        JSON.pretty_generate(
+          {
+            # JSON-style file comment
+            '//': [
+              'Animation Library Manifest',
+              'GENERATED FILE: DO NOT MODIFY DIRECTLY',
+              'See tools/scripts/rebuildAnimationLibraryManifest.rb for more information.'
+            ],
+
+            # Strip aliases from metadata - they're no longer needed since they
+            #   are represented in the alias map.
+            # Also sort for stable updates
+            'metadata': animation_metadata.hmap {|k, v| [k, v.omit!('aliases')]}.sort.to_h,
+
+            # Sort alias map for stable updates
+            'aliases': alias_map.sort.to_h
+          }
+        ),
+        acl: 'public-read',
+        no_random: true,
+        content_type: 'json'
+      )
+    end
+
+  # Report any issues while talking to S3 and suggest most likely steps for fixing it.
+  rescue Aws::Errors::ServiceError => service_error
+    warn service_error.inspect
+    warn <<-EOS.unindent
+
+      #{bold 'There was an error talking to S3.'}  Make sure you have credentials set using one of:
+
+        * aws_access_key and aws_secret_key in your locals.yml
+        * ENV['AWS_ACCESS_KEY_ID'] and ENV['AWS_SECRET_ACCESS_KEY']
+        * ~/.aws/credentials
+
+      #{dim 'See http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html for more details.'}
+    EOS
+  end
+
+  # Returns a map of names and aliases used in animation metadata
+  # for translation
+  def get_animation_strings
+    bucket = Aws::S3::Bucket.new(DEFAULT_S3_BUCKET)
+    animation_objects = get_animation_objects(bucket)
+
+    animation_metadata = build_animation_metadata(animation_objects, read_old_metadata)
+    strings = Hash.new
+    animation_metadata.each do |_, metadata|
+      next unless metadata['aliases']
+      metadata['aliases'].each do |aliaz|
+        strings[aliaz] = aliaz
```

Comment:
are we worried about collision here?

Reply:
Because the key and the value are the same, a collision won't matter. I could check for the collision but I don't know what that would get us.

Reply:
d'oh! Good point :upside_down_face: